internal/app/machined/pkg/controllers/secrets/trusted_roots.go

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package secrets
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/optional"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	"github.com/siderolabs/talos/pkg/machinery/resources/files"
+)
+
+//go:embed data/ca-certificates
+var defaultCACertificates []byte
+
+// TrustedRootsController manages CA trusted roots based on configuration.
+type TrustedRootsController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *TrustedRootsController) Name() string {
+	return "secrets.TrustedRootsController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *TrustedRootsController) Inputs() []controller.Input {
```

Comment:
hmm i guess this needs to happen way early, for eg an extension that might call out to a https server and depends on the host ca as a mount

Reply:
I guess we should at least wait for the initial CAs to be populated before starting various services

Reply:
maybe we just write the one from ca-certificates if no config is available?

Reply:
That's what this PR does, controller always writes the default one, and appends extra CAs as configured.

Reply:
ahhh I missed that

> I guess we should at least wait for the initial CAs to be populated before starting various services

i guess we can add a dependency on etcfilespecstatus?

Reply:
yep, that's the plan!